internal/checker/checker.go

```diff
@@ -17313,11 +17321,24 @@ func (c *Checker) isConstructorDeclaredThisProperty(symbol *ast.Symbol) (thisAss
 	return kind, location
 }
 
+func (c *Checker) isGlobalSymbolConstructor(node *ast.Node) bool {
+	symbol := c.getSymbolOfNode(node)
+	globalSymbol := c.getGlobalESSymbolConstructorTypeSymbolOrNil()
+	return globalSymbol != nil && symbol == globalSymbol
+}
+
 func (c *Checker) widenTypeForVariableLikeDeclaration(t *Type, declaration *ast.Node, reportErrors bool) *Type {
 	if t != nil {
+		// Currently backwards compatability with libraries that merge into `SymbolConstructor` with a `symbol` is required.
```

Comment:
This misspelling of `compatability` kind of predates me and is actually more common in the repo than the correct spelling. That being said GitHub Actions yelled at me to fix it so I have.